moveByRollPitchYawThrottleAsync Documentation changed from radians to Degrees

Gastastrophe opened the pull request:
<!-- Thank you for submitting a pull request! -->
<!-- ⚠️⚠️ Do Not Delete This! pull_request_template ⚠️⚠️ -->
<!-- Please read our contribution guidelines: https://microsoft.github.io/AirSim/CONTRIBUTING/ -->

Fixes: #    <!-- add this line for each issue your PR solves. -->
<!-- Fixes: # -->
Fixed the documentation for moveByRollPitchYawThrottleAsync to reflect that the function uses degrees instead of Radians when operating on a PX4 stack.
<!-- Fixes: # -->

## About
<!-- Describe what your PR is about. -->
After some testing, I have found that moveByRollPitchYawThrottleAsync uses degrees instead of radians for the target orientation in the body frame while simulating with HITL. The documentation says it uses radians, so I have updated it.

## How Has This Been Tested?
<!-- Please, describe how you have tested your changes to help us incorporate them. -->
I called the moveByRollPitchYawThrottleAsync function with values in ranges [-pi/2, pi/2] and [-90, 90], and I discovered that only the [-90, 90] range produced the expected changes in the drone's orientation in a PX4 HITL Unreal simulation.

## Screenshots (if appropriate):

Gastastrophe committed:
moveByRollPitchYawThrottleAsync Documentation changed from radians to degrees

Gastastrophe committed:
moveByRollPitchYawThrottleAsync Documentation changed from radians to degrees

jonyMarino commented:
@Gastastrophe very well.
This may be a difference between SimpleFlight and PX4. Could you test if the same happens using SimpleFlight?

Gastastrophe commented:
> @Gastastrophe very well. This may be a difference between SimpleFlight and PX4. Could you test if the same happens using SimpleFlight?

Good catch! I tested on SimpleFlight, and that one is in Radians. My PX4 is a Pixhawk 4 mini that I set up with QGroundControl according to the instructions, so I imagine this is a common issue for PX4 users.

jonyMarino commented:
@Gastastrophe Thanks for the test. If the Sensors and the other controllers use radians, we should change SimpleFlight to use radians. So I believe that We should convert all the pitch roll yaw parameters from [these methods](https://github.com/microsoft/AirSim/blob/f7fe3997a3457776bbe836a28a62b37b24939430/AirLib/include/vehicles/multirotor/firmwares/simple_flight/SimpleFlightApi.hpp#L213-L289) from degrees to radians. Also, We should take out all the calls to Utils::degreesToRadians. It is not a minor change; also, we should change examples and maybe more documentation. And everything needs to be tested. In the meantime, I would clarify (in the API reference that you have modified (but for all the methods that have pith, roll, and yaw parameters)) that simpleflight uses degrees, while PX4 and ardupilot (Am I right @rajat2004?) use radians. @Gastastrophe do you want to clarify that in this document so you can own your first commit?

Gastastrophe committed:
SimpleFlight and PX4 Discrepency

Updated the documentation to reflect the difference in units between SimpleFlight and PX4

Gastastrophe committed:
Revert "SimpleFlight and PX4 Discrepency"

This reverts commit 2c3b30c4ad2e420ea5a742b7b560c627477f8548.

Gastastrophe committed:
PX4 and SimpleFlight Discrepencies

Updated the documentation to reflect the difference in units reported by PX4 HITL and Simpleflight

Gastastrophe commented:
@jonyMarino I have updated the documentation and my pull request accordingly

rajat2004 commented:
ArduPilot currently doesn't support controlling moving the vehicle through the AirSim APIs hence it won't matter. Best to be consistent across firmwares, but yeah, changing this will be a big behaviour change. Maybe it can be done in a major release v2.0?
